data/refresh/rebuild_schedule_e.sql

```diff
@@ -1,31 +1,437 @@
-    image_pdf_url(image_num) as pdf_url,
-    coalesce(rpt_tp, '') in ('24', '48') as is_notice,
-    to_tsvector(pye_nm) as payee_name_text,
-    now() as pg_date
+  (cmte_id,
+ cmte_nm,
+ pye_nm,
+ payee_l_nm,
+ payee_f_nm,
+ payee_m_nm,
+ payee_prefix,
+ payee_suffix,
+ pye_st1,
+ pye_st2,
+ pye_city,
+ pye_st,
+ pye_zip,
+ entity_tp,
+ entity_tp_desc,
+ exp_desc,
+ catg_cd,
+ catg_cd_desc,
+ s_o_cand_id,
+ s_o_cand_nm,
+ s_o_cand_nm_first,
+ s_o_cand_nm_last,
+ s_o_cand_m_nm,
+ s_o_cand_prefix,
+ s_o_cand_suffix,
+ s_o_cand_office,
+ s_o_cand_office_desc,
+ s_o_cand_office_st,
+ s_o_cand_office_st_desc,
+ s_o_cand_office_district,
+ s_o_ind,
+ s_o_ind_desc,
+ election_tp,
+ fec_election_tp_desc,
+ cal_ytd_ofc_sought,
+ dissem_dt,
+ exp_amt,
+ exp_dt,
+ exp_tp,
+ exp_tp_desc,
+ memo_cd,
+ memo_cd_desc,
+ memo_text,
+ conduit_cmte_id,
+ conduit_cmte_nm,
+ conduit_cmte_st1,
+ conduit_cmte_st2,
+ conduit_cmte_city,
+ conduit_cmte_st,
+ conduit_cmte_zip,
+ indt_sign_nm,
+ indt_sign_dt,
+ notary_sign_nm,
+ notary_sign_dt,
+ notary_commission_exprtn_dt,
+ filer_l_nm,
+ filer_f_nm,
+ filer_m_nm,
+ filer_prefix,
+ filer_suffix,
+ action_cd,
+ action_cd_desc,
+ tran_id,
+ back_ref_tran_id,
+ back_ref_sched_nm,
+ schedule_type,
+ schedule_type_desc,
+ line_num,
+ image_num,
+ file_num,
+ link_id,
+ orig_sub_id,
+ sub_id,
+ filing_form,
+ rpt_tp,
+ rpt_yr,
+ election_cycle,
+ timestamp,
+ pdf_url,
+ is_notice,
+ payee_name_text,
+ pg_date
+)
+select cmte_id,
+ cmte_nm,
+ pye_nm,
+ payee_l_nm,
+ payee_f_nm,
+ payee_m_nm,
+ payee_prefix,
+ payee_suffix,
+ pye_st1,
+ pye_st2,
+ pye_city,
+ pye_st,
+ pye_zip,
+ entity_tp,
+ entity_tp_desc,
+ exp_desc,
+ catg_cd,
+ catg_cd_desc,
+ s_o_cand_id,
+ s_o_cand_nm,
+ s_o_cand_nm_first,
+ s_o_cand_nm_last,
+ s_o_cand_m_nm,
+ s_o_cand_prefix,
+ s_o_cand_suffix,
+ s_o_cand_office,
+ s_o_cand_office_desc,
+ s_o_cand_office_st,
+ s_o_cand_office_st_desc,
+ s_o_cand_office_district,
+ s_o_ind,
+ s_o_ind_desc,
+ election_tp,
+ fec_election_tp_desc,
+ cal_ytd_ofc_sought,
+ dissem_dt,
+ exp_amt,
+ exp_dt,
+ exp_tp,
+ exp_tp_desc,
+ memo_cd,
+ memo_cd_desc,
+ memo_text,
+ conduit_cmte_id,
+ conduit_cmte_nm,
+ conduit_cmte_st1,
+ conduit_cmte_st2,
+ conduit_cmte_city,
+ conduit_cmte_st,
+ conduit_cmte_zip,
+ indt_sign_nm,
+ indt_sign_dt,
+ notary_sign_nm,
+ notary_sign_dt,
+ notary_commission_exprtn_dt,
+ filer_l_nm,
+ filer_f_nm,
+ filer_m_nm,
+ filer_prefix,
+ filer_suffix,
+ action_cd,
+ action_cd_desc,
+ tran_id,
+ back_ref_tran_id,
+ back_ref_sched_nm,
+ schedule_type,
+ schedule_type_desc,
+ line_num,
+ image_num,
+ file_num,
+ link_id,
+ orig_sub_id,
+ sub_id,
+ filing_form,
+ rpt_tp,
+ rpt_yr,
+ cycle,
+ cast(null as timestamp) as timestamp,
+ image_pdf_url(image_num) as pdf_url,
+ coalesce(rpt_tp, '') in ('24', '48') as is_notice,
+ to_tsvector(pye_nm) as payee_name_text,
+ now() as pg_date
 from fec_sched_e_notice_vw;
 
-update ofec_sched_e_tmp
-set exp_dt = coalesce(exp_dt, dissem_dt);
```

Comment:
Is this not needed any more? I could not see it being handled in the `INSERT`.

Reply:
I see now that the `UPDATE` was called in 2 places. Can we take care of it in the `INSERT` itself?

Reply:
Yes, in the original script the update had been called twice so I remove one of them.  
I agree, it is better handle it in the insert statement, especially F57 does not have dissem_dt and hence not need to calculate.
I will update the script accordingly.

Reply:
This calculation had been moved to insert statement from sched_e.  
Timestamp column in ofec_sched_e had been dropped.
Queue process related objects (tables, triggers, and functions) for ofec_sched_e had been cleaned out from database since ofec_sched_e is not populated by queue process.